source/module_io/write_libxc_r.cpp

```diff
@@ -0,0 +1,285 @@
+//======================
+// AUTHOR : Peize Lin
+// DATE :   2024-09-12
+//======================
+
+#ifdef USE_LIBXC
+
+#include "write_libxc_r.h"
+#include "module_hamilt_general/module_xc/xc_functional.h"
+#include "module_hamilt_general/module_xc/xc_functional_libxc.h"
+#include "module_elecstate/module_charge/charge.h"
+#include "module_esolver/esolver_fp.h"
+#include "module_io/cube_io.h"
+#include "module_base/global_variable.h"
+#include "module_parameter/parameter.h"
+#include "module_base/timer.h"
+#include "module_base/tool_title.h"
+
+#include <xc.h>
+#include <iostream>
+#include <vector>
+#include <stdexcept>
+#include <unistd.h>
+
+void ModuleIO::write_libxc_r(

```

Comment:
could you add tests for this function? since there are many cases, I guess we need more than one test.